projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -54,30 +57,35 @@ static StatusCode prv_set_channel(Ads1015Storage *storage, Ads1015Channel channe
 // Also if there is a callback on a channel, it will be run here.
 static void prv_interrupt_handler(const GPIOAddress *address, void *context) {
   Ads1015Storage *storage = context;
+ gpio_it_mask_interrupt(&storage->ready_pin, true);
   Ads1015Channel current_channel = storage->current_channel;
   uint8_t channel_enable_bitset = storage->channel_enable_bitset;
   uint8_t read_conv_register[2] = { 0, 0 };
-
+if ((storage->channel_enable_bitset && (1 << current_channel)) != 0){
   prv_read_register(storage->i2c_port, storage->i2c_addr, ADS1015_ADDRESS_POINTER_CONV,
                     read_conv_register, SIZEOF_ARRAY(read_conv_register));
   // Following line puts the two read bytes into an int16.
-  // 4 LSB's are not part of the result hence the bitshift.
+  // 4 least significant bits are not part of the result hence the bitshift.
   storage->channel_readings[current_channel] =
       ((read_conv_register[0] << 8) | read_conv_register[1]) >> ADS1015_NUM_RESERVED_BITS_CONV_REG;
-
+ // printf("channel:%d read\n", current_channel);
+ // printf("actual:%d pending:%d\n", channel_enable_bitset, storage->channel_enable_bitset_pending);
   // Runs the users callback if not NULL.
   if (storage->channel_callback[current_channel] != NULL) {
     storage->channel_callback[current_channel](current_channel,
                                                storage->callback_context[current_channel]);
   }
+}
 
-  prv_mark_channel_enabled(current_channel, false, &storage->channel_enable_bitset_helper);
-  if (storage->channel_enable_bitset_helper == ADS1015_BITSET_EMPTY) {
-    storage->channel_enable_bitset_helper = channel_enable_bitset;
+  prv_mark_channel_enabled(current_channel, false, &storage->channel_enable_bitset_pending);
+  if (storage->channel_enable_bitset_pending == ADS1015_BITSET_EMPTY) {
+    // Reset the pending bitset once gone through a cycle of channel rotation.
+    storage->channel_enable_bitset_pending = channel_enable_bitset;
   }
-  current_channel = __builtin_ffs(storage->channel_enable_bitset_helper) - 1;
+  current_channel = __builtin_ffs(storage->channel_enable_bitset_pending) - 1;
   // Update so that the ADS1015 reads from the next channel.
   prv_set_channel(storage, current_channel);
+gpio_it_mask_interrupt(&storage->ready_pin, false);
```

Comment:
Same goes for this - why would masking the interrupt inside the interrupt have an effect?